addons/web/static/src/js/fields/abstract_field.js

```diff
@@ -299,6 +299,32 @@ var AbstractField = Widget.extend({
         return this._render() || $.when();
     },
 
+    /**
+     * add the invalid class on a field
+     */
+    setInvalidClass: function () {
```

Comment:
I know you don't like alphabetical order, and I am unsure about it, but maybe at least order these 3 methods

Reply:
done